File: src/mysqlcookbook/recipes/transfer/lookup_test.rb
#!/usr/bin/ruby
# lookup_test.rb: Demonstrate use of hash for lookup/validation purposes.

# create hash, add members to it, using values as keys

members = Hash[
  "pig"   => 1,
  "cat"   => 1,
  "duck"  => 1,
  "dog"   => 1,
]

# show list of legal values

puts "Legal values: " + members.keys.sort.join(" ")

# test some values to see whether they're in the hash

puts "Test some values:"
["cat", "snake"].each do |val|
  if members.has_key?(val)
    puts val + " (good)"
  else
    puts val + " (bad)"
  end
end

File: src/mysqlcookbook/recipes/transfer/lookup_test.php
<?php
# lookup_test.php: Demonstrate use of associative array for
# lookup/validation purposes.

# create array, add members to it, using values as keys

$members = array (
  "pig"  => 1,
  "cat"  => 1,
  "duck" => 1,
  "dog"  => 1
);

# show list of legal values

print ("Legal values: " . implode (" ", array_keys ($members)) . "\n");

# test some values to see whether they're in the array

print ("Test some values:\n");
$values = array ("cat", "snake");
foreach ($values as $val)
{
  if (isset ($members[$val]))
    print ($val . " (good)\n");
  else
    print ($val . " (bad)\n");
}
?>
